apps/userdashboard/assets/js/a4_candy_userdashboard/react_moderation_notification_list.jsx

```diff
@@ -1,22 +1,45 @@
 import ModerationNotificationList from './ModerationNotificationList'
 import React from 'react'
-import ReactDOM from 'react-dom'
+import { createRoot } from 'react-dom/client'
+import { widget as ReactWidget } from 'adhocracy4'
 
 function init () {
-  $('[data-aplus-widget="moderation_notification_list"]').each(function (i, element) {
-    const moderationCommentsApiUrl = element.getAttribute('data-moderation-comments-api-url')
-    const projectTitle = element.getAttribute('data-project-title')
-    const organisation = element.getAttribute('data-organisation')
-    const projectUrl = element.getAttribute('data-project-url')
-    ReactDOM.render(
-      <ModerationNotificationList
-        moderationCommentsApiUrl={moderationCommentsApiUrl}
-        projectTitle={projectTitle}
-        organisation={organisation}
-        projectUrl={projectUrl}
-      />,
-      element)
-  })
+  ReactWidget.initialise('aplus', 'moderation_notification_list',
+    function (el) {
+      const moderationCommentsApiUrl = el.getAttribute('data-moderation-comments-api-url')
+      const projectTitle = el.getAttribute('data-project-title')
+      const organisation = el.getAttribute('data-organisation')
+      const projectUrl = el.getAttribute('data-project-url')
+      const root = createRoot(el)
+      root.render(
+        <ModerationNotificationList
+          moderationCommentsApiUrl={moderationCommentsApiUrl}
+          projectTitle={projectTitle}
+          organisation={organisation}
+          projectUrl={projectUrl}
+        />)
+    })
 }
 
 document.addEventListener('DOMContentLoaded', init, false)
+
+// function init () {
+//   ReactWidget.initialise('aplus', 'moderation_notification_list',
+//     function (el) {
+//       const props = JSON.parse(el.getAttribute('data-attributes'))
```

Comment:
@goapunk @khamui this doesn't work and i'm not sure why, it's more a point of curiosity then anything but wanted to ask if there's anything i'm missing as it works in other apps, when this is used then props is empty, however it will work with `const props = JSON.parse(JSON.stringify(el.dataset))` not sure why but as dataset is meant for less complicated data structures I though I shouldn't introduce it, anyway will finish off removing excess js

Reply:
I'm not sure I follow. The `moderation_notification_list` doesn't have a `data-attributes`, so it also doesn't find one here so I assume it's a typo :) what works is `const props = el.dataset`, no need to parse

Reply:
@goapunk not a typo just me being dumb, thought data-attributes should fetch all the data-*- attributes. Anyway will swap to using dataset then if are not worried about introducing new syntax, cheers!